WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift

```diff
@@ -44,18 +48,34 @@ final class RefundConfirmationViewModel {
         )
     ]
 
-    init(details: Details, currencySettings: CurrencySettings = ServiceLocator.currencySettings) {
+    init(details: Details, actionProcessor: StoresManager = ServiceLocator.stores, currencySettings: CurrencySettings = ServiceLocator.currencySettings) {
```

Comment:
I like this idiom. We get the best of both worlds. Having global dependencies clearly defined by a global entity and being able to inject dependencies without factories and local DI via the constructor without adding the rigidity that DI tends to add to all designs. Nice.

Reply:
(Sorry, that was just me thinking out loud 😊)

Reply:
😄